tools/ansible/wsu.yaml

```diff
@@ -0,0 +1,135 @@
+             wmcb_location is not defined or
+             openshift_node_bootstrap_server is not defined
+
+    - name: Show ignition file endpoint 
+      debug:
+        msg: "Ignition file endpoint is {{ openshift_node_bootstrap_endpoint }}"
+
+    - name: Create temporary pod template file
+      file:
+        path: curl_pod.yaml
+        state: touch
+        mode: '0644'
+
+    - name: Copy pod template to temporary pod file
+      copy:
+        src: curl_pod_template.yaml
+        dest: curl_pod.yaml
+        mode: '0644'
+
+    - name: Change endpoint location in temporary pod template entrypoint 
+      replace:
+        path: curl_pod.yaml
+        regexp: "<endpoint>"
+        replace: "{{ openshift_node_bootstrap_endpoint }}" 
+
+    - name: Run pod on Cluster
+      command: oc apply -f curl_pod.yaml
+
+    - name: Wait until pod is created 
+      command: oc get pod curl-ign-pod
+      register: pod_status
+      delay: 5
+      retries: 10
+      until:
+        - '"ContainerCreating" not in pod_status.stdout'
+
+    - name: Create temporary build directory to store files
+      tempfile:
+        state: directory
+      register: build_tmp
+
+    - name: Show temporary path directory
+      debug:
+        msg: "Temporary build path is {{ build_tmp.path }}"
+      when: build_tmp is defined
+
+    - name: Fetch cluster ignition file to controller machine
+      shell: oc logs curl-ign-pod > "{{ build_tmp.path }}/worker.ign"
+      when: build_tmp is defined
+
+    - name: Download Kubernetes archive to controller machine
+      get_url:
+        url: "{{ kubelet_location }}"
+        dest: "{{ build_tmp.path }}/kube.tar.gz"
+        mode: 0400
+      when: build_tmp is defined
+
+    - name: Unarchive Kubenetes binary
+      unarchive:
+        src: "{{ build_tmp.path }}/kube.tar.gz"
+        dest: "{{ build_tmp.path }}" 
+      when: build_tmp is defined
+
+    - name: Download wmcb binary to controller machine
+      get_url:
+        url: "{{ wmcb_location }}"
+        dest: "{{ build_tmp.path }}/wmcb.exe"
+        mode: 0400
+      when: build_tmp is defined
+
+    - name: Change temporary path permission
+      file:
+        path: "{{ build_tmp.path }}"
+        mode: '0755'
+        recurse: yes 
+      when: build_tmp is defined
+
+    - name: Remove pod
+      command: oc delete pod curl-ign-pod
+
+    - name: Remove temporary pod template
+      file:
+        path: curl_pod.yaml
+        state: absent
+
+
+- hosts: win
+  vars:
+    win_build_tmp: "{{ hostvars['localhost']['build_tmp'] }}"
+  tasks:
+    - name: Description 
+      debug:
+        msg: Preparing your Windows node 
+
+    - name: Check Windows container runtime
+      win_service:
+        name: docker
+        state: started
+
+    - name: Transfer kubelet binary from controller machine to Windows host
+      win_copy: 
+        src: "{{ win_build_tmp.path }}/kubernetes/node/bin/kubelet.exe"
+        dest: C:\Windows\Temp\kubelet.exe
```

Comment:
Why can't you create a new temp directory with just our files instead of hard coding the location?

Reply:
Does it need to be a temporary directory which deleted after the playbook ran, or a custom temporary directory that only store bootstrap files?

Reply:
Looking at this again, how can we keep these files in a temporary location that could get removed? What happens if we reboot the instance and `wmcb` needs to run again? We needs files to be persisted,

Reply:
Since we're going to run the Kubelet, OVN as services rebooting shouldn't cause a problem? If someone needs to `WMCB` again, they should use the playbook again.

Reply:
Ah yes, I forget running wmcb is a one time operation.

But @leonjia0112 the question about creating a temp directory still stands. We cannot assume `C:\Windows\Temp` always exists.